registry/storage/registry.go

```diff
@@ -12,28 +12,60 @@ import (
 // package. All instances should descend from this object.
 type registry struct {
 	blobStore                   *blobStore
-	blobServer                  distribution.BlobServer
-	statter                     distribution.BlobStatter // global statter service.
+	blobServer                  *blobServer
```

Comment:
The goal with this is to actually get them separated by this interfaces in a clean manner. Using an interface declares that only these operations should be available, reducing the temptation to reach down into the object and increate coupling (this section was particularly intertwined before). For example, it took some work to par down blobServer to just the `BlobServer` interface. I feel like we can afford an extra type assertion if it helps maintain separation.

Granted, we do need a better way to push these kinds of options up and down the stack. Looking at `BlobDescriptorCacheProvider` hints that we need to adjust the way cache instantiation is done. You should have to only set `registry.blobDescriptorCacheProvider` and then the other objects should be automatic populate with the new `statter`. This is probably outside the scope of this PR.